Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm

```diff
@@ -708,13 +706,58 @@ static bool needsEAGLOnMac()
     return std::make_tuple(eglImage, textureWidth, textureHeight);
 }
 
-void GraphicsContextGLCocoa::detachIOSurfaceFromSharedTexture(void* handle)
+GraphicsContextGL::ExternalImageAttachment GraphicsContextGLCocoa::attachIOSurfaceToExternalImage(GCGLenum target, IOSurface* surface)
 {
-    auto display = platformDisplay();
-    EGL_DestroyImageKHR(display, handle);
+    MTLSharedTextureHandle* handle = [[MTLSharedTextureHandle alloc] initWithIOSurface:surface->surface() label:@"WebXR"];
+    if (!handle) {
+        LOG(WebGL, "Unable to create a MTLSharedTextureHandle from the IOSurface in attachIOSurfaceToExternalImage.");
+        return std::nullopt;
+    }
+
+    return attachMTLTextureToExternalImage(platformDisplay(), target, handle);
 }
 #endif
 
+GraphicsContextGL::ExternalImageAttachment GraphicsContextGLCocoa::attachExternalImage(GCGLenum target, GCGLExternalImage type, WTF::MachSendRight imageSendRight)
+{
+#if PLATFORM(IOS_FAMILY_SIMULATOR)
+    // MTLSharedTexture is unsupported on simulator
+    UNUSED_VARIABLE(target);
+    UNUSED_VARIABLE(type);
+    UNUSED_VARIABLE(imageSendRight);
+
+    return std::nullopt;
+#else // !PLATFORM(IOS_FAMILY_SIMULATOR
+
+    MTLSharedTextureHandle* handle = nil;
+    switch (type) {
+    case GCGLExternalImage::IOSurface: {
+        auto surface = IOSurface::createFromSendRight(WTFMove(imageSendRight));
+        if (!surface) {
+            LOG(WebGL, "Unable to create an IOSurface from imagePort in attachExternalImage.");
```

Comment:
maybe we could remove the excessive logging. The log messages are already not consistent with the code.

Reply:
I was copying the existing style; when in doubt make it look like what was there.

I agree that the logging is inconsistent with the rest of the code.

Reply:
With inconsistency, I meant that it refers to, for example, "imagePort" which doesn't exist. Feel free to choose, match the existing logging or not.